--- simavr/Makefile.common.orig	2015-07-29 15:31:40.393146823 +1000
+++ simavr/Makefile.common	2015-07-29 15:39:15.502335219 +1000
@@ -45,7 +45,7 @@
 
 ifeq (${shell uname}, Darwin)
  # gcc 4.2 from MacOS is really not up to scratch anymore 
- CC			= clang
+ CC		:= clang
  AVR_ROOT 	:= "/Applications/Arduino.app/Contents/Resources/Java/hardware/tools/avr/"
  AVR_INC 	:= ${AVR_ROOT}/avr/
  AVR 		:= ${AVR_ROOT}/bin/avr-
