--- smatch/Makefile.orig	2015-07-29 11:08:38.441896696 +1000
+++ smatch/Makefile	2015-07-29 15:12:01.345521886 +1000
@@ -8,14 +8,23 @@
 $(shell echo '#define SPARSE_VERSION "$(SPARSE_VERSION)"' > version.h)
 endif
 
-OS = linux
+ifeq ($(OS),Windows_NT)
+OS = mingw
+else
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Darwin)
+OS = bsd
+else
+OS = linux 
+endif
+endif
 
 ifeq ($(CC),"")
 CC = gcc
 endif
-CFLAGS = -O2 -finline-functions -fno-strict-aliasing -g
-CFLAGS += -Wall -Wwrite-strings
-LDFLAGS += -g -lm -lsqlite3
+CFLAGS = -O2 -finline-functions -fno-strict-aliasing -flto
+CFLAGS += -Wall -Wwrite-strings -I$(PREFIX)/include
+LDFLAGS += -lm -lsqlite3
 LD = gcc
 AR = ar
 PKG_CONFIG = pkg-config
@@ -47,7 +56,6 @@
 endif
 
 DESTDIR=
-PREFIX=$(HOME)
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/lib
 MANDIR=$(PREFIX)/share/man
@@ -211,7 +219,7 @@
 	$(QUIET_LINK)$(LD) -o $@ $^ $($@_EXTRA_OBJS) $(LDFLAGS)
 
 smatch: smatch.o $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) 
-	$(CC) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) $(LDFLAGS)
+	$(CC) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) $(LDFLAGS) -lsqlite3
 
 $(LIB_FILE): $(LIB_OBJS)
 	$(QUIET_AR)$(AR) rcs $@ $(LIB_OBJS)
