diff -ur simavr.orig/simavr/Makefile simavr/simavr/Makefile
--- simavr.orig/simavr/Makefile	2012-03-12 05:25:55.428170133 +1100
+++ simavr/simavr/Makefile	2012-03-13 22:56:38.129020512 +1100
@@ -21,7 +21,7 @@
 
 target	= run_avr
 
-CFLAGS	+= -Werror
+#CFLAGS	+= -Werror
 # tracing id useful especialy if you develop simavr core.
 # it otherwise eat quite a bit of few cycles, even disabled
 #CFLAGS	+= -DCONFIG_SIMAVR_TRACE=1
