--- ../download/simavr/examples/Makefile.opengl	2012-06-18 19:44:47.236652526 +1000
+++ simavr/examples/Makefile.opengl	2012-06-18 20:58:44.440598329 +1000
@@ -8,10 +8,14 @@
 CPPFLAGS	+= ${shell pkg-config --cflags glu}
 LDFLAGS 	+= ${shell pkg-config --libs glu} -lglut
 else
+ifeq (${UNAME}, MINGW32_NT-6.1)
+LDFLAGS += -mwindows ${PREFIX}/lib/glut32.lib -lopengl32 -lglut32
+else
 CPPFLAGS	+= ${shell pkg-config --cflags glu glut} -DFREEBSD=1
 LDFLAGS 	+= ${shell pkg-config --libs glu glut}
 endif
 endif
+endif
 
 ifeq  (${UNAME}, FreeBSD)
 CPPFLAGS	+= -DFREEBSD=1 -DNO_ALLOCA=1
diff -ur ../download/simavr/Makefile.common simavr/Makefile.common
--- ../download/simavr/Makefile.common	2012-03-11 10:53:04 +1100
+++ simavr/Makefile.common	2012-03-11 10:56:13 +1100
@@ -49,8 +49,16 @@
 AVR_ROOT 	:= /usr/lib/avr
 AVR_INC 	:= ${AVR_ROOT}
 AVR 		:= avr-
+ifeq (${shell uname}, MINGW32_NT-6.1)
+AVR_ROOT    := ${PREFIX}
+AVR_INC     := ${AVR_ROOT}/avr
+AVR         := ${AVR_ROOT}/bin/avr-
+IPATH       += ${PREFIX}/include
+CFLAGS      += -I${PREFIX}/include
+else
 CFLAGS 		+= -fPIC
 endif
+endif
 
 CC 			?= gcc
 AR 			?= ar
@@ -67,6 +75,11 @@
 
 LDFLAGS 	+= -lelf 
 
+ifeq (${shell uname}, MINGW32_NT-6.1)
+LDFLAGS      += -L"/c/Program Files (x86)/Microsoft SDKs/Windows/v7.0a/lib/ws2_32.lib"
+LDFLAGS      += -lws2_32
+endif
+
 ifeq (${shell uname}, Linux)
 # allow the shared library to be found in the build directory
 LFLAGS		+= -Wl,-rpath,${LIBDIR}
@@ -86,6 +99,19 @@
 
 # --mcall-prologues can be used here, but messes up debugging a little
 %.axf: %.c 
+ifeq ($(V),1)
+	part=${<} ; part=$${part/_*}; \
+	${AVR}gcc -Wall -gdwarf-2 -Os -std=gnu99 \
+			-mmcu=$$part \
+			-DF_CPU=8000000 \
+			-fno-inline-small-functions \
+			-ffunction-sections -fdata-sections \
+			-Wl,--relax,--gc-sections \
+			-Wl,--undefined=_mmcu,--section-start=.mmcu=0x910000 \
+			-I../include -I../../include \
+			${^} -o ${@}
+	${AVR}size ${@}|sed '1d'    
+else
 	@echo AVR-CC ${<}
 	@part=${<} ; part=$${part/_*}; \
 	${AVR}gcc -Wall -gdwarf-2 -Os -std=gnu99 \
@@ -98,6 +124,7 @@
 			-I../include -I../../include \
 			${^} -o ${@}
 	@${AVR}size ${@}|sed '1d'
+endif
 
 # this rule has precedence
 ${OBJ}/sim_%.o : cores/sim_%.c
diff -ur ../download/simavr/examples/board_hd77480/charlcd.c simavr/examples/board_hd77480/charlcd.c
--- ../download/simavr/examples/board_hd77480/charlcd.c	2012-03-11 10:53:03 +1100
+++ simavr/examples/board_hd77480/charlcd.c	2012-03-11 10:52:48 +1100
@@ -33,6 +33,9 @@
 #if __APPLE__
 #include <GLUT/glut.h>
 #else
+#ifdef __MINGW32__
+#include <windows.h>
+#endif
 #include <GL/glut.h>
 #endif
 #include <pthread.h>
diff -ur ../download/simavr/examples/board_ledramp/ledramp.c simavr/examples/board_ledramp/ledramp.c
--- ../download/simavr/examples/board_ledramp/ledramp.c	2012-03-11 10:53:03 +1100
+++ simavr/examples/board_ledramp/ledramp.c	2012-03-11 10:52:48 +1100
@@ -25,6 +25,9 @@
 #if __APPLE__
 #include <GLUT/glut.h>
 #else
+#ifdef __MINGW32__
+#include <windows.h>
+#endif
 #include <GL/glut.h>
 #endif
 #include <pthread.h>
diff -ur ../download/simavr/examples/board_timer_64led/timer_64led.c simavr/examples/board_timer_64led/timer_64led.c
--- ../download/simavr/examples/board_timer_64led/timer_64led.c	2012-03-11 10:53:03 +1100
+++ simavr/examples/board_timer_64led/timer_64led.c	2012-03-11 10:52:48 +1100
@@ -25,6 +25,9 @@
 #if __APPLE__
 #include <GLUT/glut.h>
 #else
+#ifdef __MINGW32__
+#include <windows.h>
+#endif
 #include <GL/glut.h>
 #endif
 #include <pthread.h>
diff -ur ../download/simavr/examples/parts/hd44780_glut.c simavr/examples/parts/hd44780_glut.c
--- ../download/simavr/examples/parts/hd44780_glut.c	2012-03-11 10:53:04 +1100
+++ simavr/examples/parts/hd44780_glut.c	2012-03-11 10:52:49 +1100
@@ -25,6 +25,9 @@
 #if __APPLE__
 #include <GLUT/glut.h>
 #else
+#ifdef __MINGW32__
+#include <windows.h>
+#endif
 #include <GL/glut.h>
 #endif
 
diff -ur ../download/simavr/examples/parts/uart_udp.c simavr/examples/parts/uart_udp.c
--- ../download/simavr/examples/parts/uart_udp.c	2012-03-11 10:53:04 +1100
+++ simavr/examples/parts/uart_udp.c	2012-03-11 10:52:49 +1100
@@ -19,10 +19,15 @@
 	along with simavr.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef __MINGW32__
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <sys/socket.h>
+#endif
 #include <sys/time.h>
 #include <pthread.h>
 #include <string.h>
@@ -97,7 +102,7 @@
 			continue;
 
 		if (FD_ISSET(p->s, &read_set)) {
-			uint8_t buffer[512];
+			char buffer[512];
 
 			socklen_t len = sizeof(p->peer);
 			ssize_t r = recvfrom(p->s, buffer, sizeof(buffer)-1, 0, (struct sockaddr*)&p->peer, &len);
@@ -105,16 +110,21 @@
 		//	hdump("udp recv", buffer, r);
 
 			// write them in fifo
-			uint8_t * src = buffer;
+			char * src = buffer;
 			while (r-- && !uart_udp_fifo_isfull(&p->out))
 				uart_udp_fifo_write(&p->out, *src++);
-			if (r > 0)
+			if (r > 0) {
+#if __MINGW32__
+				printf("UDP dropped %I64u bytes\n", (unsigned long long)r);
+#else
 				printf("UDP dropped %zu bytes\n", r);
+#endif
+			}
 		}
 		if (FD_ISSET(p->s, &write_set)) {
-			uint8_t buffer[512];
+			char buffer[512];
 			// write them in fifo
-			uint8_t * dst = buffer;
+			char * dst = buffer;
 			while (!uart_udp_fifo_isempty(&p->in) && dst < (buffer+sizeof(buffer)))
 				*dst++ = uart_udp_fifo_read(&p->in);
 			socklen_t len = dst - buffer;
diff -ur ../download/simavr/examples/parts/uart_udp.h simavr/examples/parts/uart_udp.h
--- ../download/simavr/examples/parts/uart_udp.h	2012-03-11 10:53:04 +1100
+++ simavr/examples/parts/uart_udp.h	2012-03-11 10:52:49 +1100
@@ -23,7 +23,12 @@
 #ifndef __UART_UDP_H___
 #define __UART_UDP_H___
 
+#ifdef __MINGW32__
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
 #include <netinet/in.h>
+#endif
 #include "sim_irq.h"
 #include "fifo_declare.h"
 
diff -ur ../download/simavr/simavr/Makefile simavr/simavr/Makefile
--- ../download/simavr/simavr/Makefile	2012-03-11 10:53:05 +1100
+++ simavr/simavr/Makefile	2012-03-11 10:52:50 +1100
@@ -43,7 +43,10 @@
 IPATH	+= .
 IPATH	+= ../../shared
 IPATH	+= ../include
-    
+ifeq (${shell uname}, MINGW32_NT-6.1)
+IPATH       += ${PREFIX}/include
+endif
+
 #
 # Static library
 #
diff -ur ../download/simavr/simavr/cores/sim_tiny13.c simavr/simavr/cores/sim_tiny13.c
--- ../download/simavr/simavr/cores/sim_tiny13.c	2012-03-11 10:53:05 +1100
+++ simavr/simavr/cores/sim_tiny13.c	2012-03-11 10:52:49 +1100
@@ -20,7 +20,7 @@
 	along with simavr.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include </usr/include/stdio.h>
+#include <stdio.h>
 #include "sim_avr.h"
 #include "sim_core_declare.h"
 #include "avr_eeprom.h"
diff -ur ../download/simavr/simavr/cores/sim_tiny2313.c simavr/simavr/cores/sim_tiny2313.c
--- ../download/simavr/simavr/cores/sim_tiny2313.c	2012-03-11 10:53:05 +1100
+++ simavr/simavr/cores/sim_tiny2313.c	2012-03-11 10:52:49 +1100
@@ -19,7 +19,7 @@
 	along with simavr.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include </usr/include/stdio.h>
+#include <stdio.h>
 #include "sim_core_declare.h"
 #include "avr_eeprom.h"
 #include "avr_watchdog.h"
Only in simavr/simavr: obj-
diff -ur ../download/simavr/simavr/sim/avr_watchdog.h simavr/simavr/sim/avr_watchdog.h
--- ../download/simavr/simavr/sim/avr_watchdog.h	2012-03-11 10:53:05 +1100
+++ simavr/simavr/sim/avr_watchdog.h	2012-03-11 10:52:50 +1100
@@ -68,7 +68,7 @@
 /* no WDP3, WDIE, WDIF in atmega128 */
 #define AVR_WATCHDOG_DECLARE_128(_WDSR, _vec) \
 	.watchdog = {\
-		.wdrf = AVR_IO_REGBIT(MCUSR, WDRF),\
+		.wdrf = AVR_IO_REGBIT(MCUCSR, WDRF),\
 		.wdce = AVR_IO_REGBIT(_WDSR, WDCE),\
 		.wde = AVR_IO_REGBIT(_WDSR, WDE),\
 		.wdp = { AVR_IO_REGBIT(_WDSR, WDP0),AVR_IO_REGBIT(_WDSR, WDP1),\
diff -ur ../download/simavr/simavr/sim/sim_elf.c simavr/simavr/sim/sim_elf.c
--- ../download/simavr/simavr/sim/sim_elf.c	2012-03-11 10:53:06 +1100
+++ simavr/simavr/sim/sim_elf.c	2012-03-11 10:52:50 +1100
@@ -25,9 +25,11 @@
 
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifndef __MINGW32__
 #include <unistd.h>
-#include <stdlib.h>
 #include <stdio.h>
+#endif
+#include <stdlib.h>
 #include <string.h>
 #include <libelf.h>
 #include <gelf.h>
@@ -280,12 +282,20 @@
 	//	hdump("code", data_text->d_buf, data_text->d_size);
 		memcpy(firmware->flash + offset, data_text->d_buf, data_text->d_size);
 		offset += data_text->d_size;
+#ifdef __MINGW32__
+		printf("Loaded %Iu .text\n", data_text->d_size);
+#else
 		printf("Loaded %zu .text\n", data_text->d_size);
+#endif
 	}
 	if (data_data) {
 	//	hdump("data", data_data->d_buf, data_data->d_size);
 		memcpy(firmware->flash + offset, data_data->d_buf, data_data->d_size);
+#ifdef __MINGW32__
+		printf("Loaded %Iu .data\n", data_data->d_size);
+#else
 		printf("Loaded %zu .data\n", data_data->d_size);
+#endif
 		offset += data_data->d_size;
 		firmware->datasize = data_data->d_size;
 	}
@@ -293,7 +303,11 @@
 	//	hdump("eeprom", data_ee->d_buf, data_ee->d_size);
 		firmware->eeprom = malloc(data_ee->d_size);
 		memcpy(firmware->eeprom, data_ee->d_buf, data_ee->d_size);
+#ifdef __MINGW32__
+		printf("Loaded %Iu .eeprom\n", data_ee->d_size);
+#else
 		printf("Loaded %zu .eeprom\n", data_ee->d_size);
+#endif
 		firmware->eesize = data_ee->d_size;
 	}
 //	hdump("flash", avr->flash, offset);
diff -ur ../download/simavr/simavr/sim/sim_gdb.c simavr/simavr/sim/sim_gdb.c
--- ../download/simavr/simavr/sim/sim_gdb.c	2012-03-11 10:53:06 +1100
+++ simavr/simavr/sim/sim_gdb.c	2012-03-11 10:52:51 +1100
@@ -19,17 +19,22 @@
 	along with simavr.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef __MINGW32__
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <sys/socket.h>
+#include <poll.h>
+#endif
 #include <sys/time.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
 #include <errno.h>
-#include <poll.h>
 #include <pthread.h>
 #include "sim_avr.h"
 #include "sim_hex.h"
@@ -64,7 +69,11 @@
 	}
 	sprintf((char*)dst, "#%02x", check);
 	DBG(printf("%s '%s'\n", __FUNCTION__, reply);)
+#if __MINGW32__
+	send(g->s, (const char *)reply, dst - reply + 3, 0);
+#else
 	send(g->s, reply, dst - reply + 3, 0);
+#endif
 }
 
 static void gdb_send_quick_status(avr_gdb_t * g, uint8_t signal)
@@ -309,24 +318,26 @@
 
 	if (ret == 0)
 		return 0;
-	
+
 	if (FD_ISSET(g->listen, &read_set)) {
 		g->s = accept(g->listen, NULL, NULL);
 
 		if (g->s == -1) {
 			perror("gdb_network_handler accept");
+#ifndef __MINGW32__
 			sleep(5);
+#endif
 			return 1;
 		}
-        int i = 1;
-        setsockopt (g->s, IPPROTO_TCP, TCP_NODELAY, &i, sizeof (i));
+		char i = 1;
+		setsockopt (g->s, IPPROTO_TCP, TCP_NODELAY, &i, sizeof (i));
 		g->avr->state = cpu_Stopped;
		printf("%s connection opened\n", __FUNCTION__);		
 	}
		
 	if (g->s != -1 && FD_ISSET(g->s, &read_set)) {
-		uint8_t buffer[1024];
-		
+		char buffer[1024];
+
 		ssize_t r = recv(g->s, buffer, sizeof(buffer)-1, 0);
 
 		if (r == 0) {
@@ -339,14 +350,16 @@
 		}
 		if (r == -1) {
 			perror("gdb_network_handler recv");
+#ifndef __MINGW32__
 			sleep(1);
+#endif
 			return 1;
 		}
 		buffer[r] = 0;
 	//	printf("%s: received %d bytes\n'%s'\n", __FUNCTION__, r, buffer);
 	//	hdump("gdb", buffer, r);
 
-		uint8_t * src = buffer;
+		char * src = buffer;
 		while (*src == '+' || *src == '-')
 			src++;
 		// control C -- lets send the guy a nice status packet
@@ -357,7 +370,7 @@
 		}
 		if (*src  == '$') {
 			// strip checksum
-			uint8_t * end = buffer + r - 1;
+			char * end = buffer + r - 1;
 			while (end > src && *end != '#')
 				*end-- = 0;
 			*end = 0;
@@ -407,7 +420,7 @@
 		return -1;
 	}
 
-	int i = 1;
+	char i = 1;
 	setsockopt(g->listen, SOL_SOCKET, SO_REUSEADDR, &i, sizeof(i));
 
 	struct sockaddr_in address = { 0 };
diff -ur ../download/simavr/tests/tests.c simavr/tests/tests.c
--- ../download/simavr/tests/tests.c	2012-03-11 10:53:06 +1100
+++ simavr/tests/tests.c	2012-03-11 10:52:51 +1100
@@ -23,8 +23,10 @@
 }
 
 void tests_success(void) {
+#ifndef __MINGW32__
 	if (orig_stderr)
 		stderr = orig_stderr;
+#endif
 	fprintf(stderr, "OK: %s\n", test_name);
 	finished = 1;
 }
@@ -95,11 +97,13 @@
 
 avr_t *tests_init_avr(const char *elfname) {
 	tests_cycle_count = 0;
+#ifndef __MINGW32__
 	if (tests_disable_stdout) {
 		orig_stderr = stderr;
 		fclose(stdout);
 		stderr = stdout;
 	}
+#endif
 	elf_firmware_t fw;
 	if (elf_read_firmware(elfname, &fw))
 		fail("Failed to read ELF firmware \"%s\"", elfname);
@@ -228,8 +232,10 @@
 }
 
 void _fail(const char *filename, int linenum, const char *fmt, ...) {
+#ifndef __MINGW32__
 	if (orig_stderr)
 		stderr = orig_stderr;
+#endif
 
 	if (filename)
 		fprintf(stderr, "%s:%d: ", filename, linenum);
--- ../download/simavr/simavr/sim/sim_avr.c	2012-05-12 10:11:23 +1000
+++ simavr/simavr/sim/sim_avr.c	2012-05-12 16:21:02 +1000
@@ -22,7 +22,9 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#ifndef __MINGW32__
 #include <unistd.h>
+#endif
 #include "sim_avr.h"
 #include "sim_core.h"
 #include "sim_time.h"
@@ -237,6 +239,28 @@
 
 }
 
+#ifdef __MINGW32__
+#include <windows.h>
+/* Borrowed from ScummVM under the GPL
+ * http://sourceforge.net/apps/trac/scummvm/browser/vendor/freesci/glutton/src/win32/usleep.c?rev=38187
+ */
+void usleep (long usec) {
+	LARGE_INTEGER lFrequency;
+	LARGE_INTEGER lEndTime;
+	LARGE_INTEGER lCurTime;
+
+	QueryPerformanceFrequency (&lFrequency);
+	if (lFrequency.QuadPart) {
+		QueryPerformanceCounter (&lEndTime);
+		lEndTime.QuadPart += (LONGLONG) usec * lFrequency.QuadPart / 1000000;
+		do {
+			QueryPerformanceCounter (&lCurTime);
+			Sleep(0);
+		} while (lCurTime.QuadPart < lEndTime.QuadPart);
+	}
+}
+#endif
+
 void avr_callback_sleep_raw(avr_t * avr, avr_cycle_count_t howLong)
 {
 	uint32_t usec = avr_cycles_to_usec(avr, howLong);
--- simavr/simavr/sim/sim_vcd_file.c.orig	2012-05-12 10:11:23 +1000
+++ simavr/simavr/sim/sim_vcd_file.c	2012-05-12 16:25:43 +1000
@@ -111,7 +111,11 @@
 			
 		if (base > oldbase || li == 0) {
 			seen = 0;
+#if __MINGW32__
+			fprintf(vcd->output, "#%I64u\n", base);
+#else		
 			fprintf(vcd->output, "#%llu\n", (long long unsigned int)base);
+#endif
 			oldbase = base;
 		}
 		seen |= (1 << l->signal->irq.irq);	// mark this trace as seen for this timestamp
