--- smatch.orig/Makefile	2010-10-29 14:05:56 +1100
+++ smatch/Makefile	2010-10-29 14:05:22 +1100
@@ -1,11 +1,11 @@
 VERSION=0.4.2
 
-OS = linux
+OS = mingw
 
 CC = gcc
-CFLAGS = -O2 -finline-functions -fno-strict-aliasing -g
-CFLAGS += -Wall -Wwrite-strings
-LDFLAGS += -g -lm -lsqlite3
+CFLAGS = -O2 -finline-functions -fno-strict-aliasing
+CFLAGS += -Wall -Wwrite-strings -I$(PREFIX)/include
+LDFLAGS += -lm -lsqlite3
 AR = ar
 
 #
@@ -27,7 +27,6 @@
 endif
 
 DESTDIR=
-PREFIX=$(HOME)
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/lib
 MANDIR=$(PREFIX)/share/man
@@ -151,10 +150,10 @@
 
 $(foreach p,$(PROGRAMS),$(eval $(p): $($(p)_EXTRA_DEPS) $(LIBS)))
 $(PROGRAMS): % : %.o 
-	$(QUIET_LINK)$(CC) $(LDFLAGS) -o $@ $^ $($@_EXTRA_OBJS) 
+	$(QUIET_LINK)$(CC) $(LDFLAGS) -o $@ $^ $($@_EXTRA_OBJS)
 
 smatch: smatch.o $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) 
-	$(CC) $(LDFLAGS) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) 
+	$(CC) $(LDFLAGS) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) -lsqlite3
 
 $(LIB_FILE): $(LIB_OBJS)
 	$(QUIET_AR)$(AR) rcs $@ $(LIB_OBJS)
