--- smatch/Makefile.orig	2012-05-12 11:16:10 +1000
+++ smatch/Makefile	2012-05-12 13:45:37 +1000
@@ -1,11 +1,11 @@
 VERSION=0.4.3
 
-OS = linux
+OS = mingw
 
 CC = gcc
-CFLAGS = -O2 -finline-functions -fno-strict-aliasing -g
-CFLAGS += -Wall -Wwrite-strings
-LDFLAGS += -g -lm -lsqlite3
+CFLAGS = -O2 -finline-functions -fno-strict-aliasing -flto
+CFLAGS += -Wall -Wwrite-strings -I$(PREFIX)/include
+LDFLAGS += -lm -lsqlite3
 AR = ar
 
 #
@@ -27,7 +27,6 @@
 endif
 
 DESTDIR=
-PREFIX=$(HOME)
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/lib
 MANDIR=$(PREFIX)/share/man
@@ -155,7 +154,7 @@
 	$(QUIET_LINK)$(CC) -o $@ $^ $($@_EXTRA_OBJS) $(LDFLAGS)
 
 smatch: smatch.o $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) 
-	$(CC) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) $(LDFLAGS)
+	$(CC) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) $(LDFLAGS) -lsqlite3
 
 $(LIB_FILE): $(LIB_OBJS)
 	$(QUIET_AR)$(AR) rcs $@ $(LIB_OBJS)
